frontend/src/components/UserProfileForm.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect } from 'react';
+
+const UserProfileForm = () => {
+  const [userData, setUserData] = useState({
+    user_id: "", // Primary key
+    email: "",
+    role: "", // VOLUN or ADMIN or INSTR
+    password: "",
+    created_at: "",
+  });
+
+  const [volunteerData, setVolunteerData] = useState({
+    volunteer_id: "", // Primary key
+    user_id: "", // Foreign key
+    l_name: "",
+    f_name: "",
+    total_hours: "",
+    class_preferences: "",
+    bio: "",
+    active: false,
+  });
+
+  useEffect(() => {
+    // Fetch user profile data from the backend when the component mounts
+    const fetchUserData = async () => {
+      try {
+        const response = await fetch('http://localhost:3001/api/sudo_get_USER?user_id=' + userData.user_id);
+        const data = await response.json();
+        setUserData(data);
+      } catch (error) {
+        console.error('Error fetching user data:', error);
+      }
+    };
+
+    const fetchVolunteerData = async () => {
+      try {
+        const response = await fetch('http://localhost:3001/api/sudo_get_VOLUN?user_id=' + userData.user_id);
```

Comment:
make the api call in a volunteerService.js file in frontend/src/api , then call the function here